Nightly as-if-std is failing on some targets with unused import `std::*`

Specifically `ubuntu-20.04, nightly`, `macos-latest, nightly` and `x86_64-fortanix-unknown-sgx`. For [example](https://github.com/rust-lang/backtrace-rs/actions/runs/6755931955/job/18364864125?pr=573):

```
Run cargo build --manifest-path crates/as-if-std/Cargo.toml --no-default-features
  
   Compiling as-if-std v0.1.0 (/Users/runner/work/backtrace-rs/backtrace-rs/crates/as-if-std)
error: unused import: `std::*`
  --> crates/as-if-std/src/lib.rs:14:13
   |
14 |     pub use std::*;
   |             ^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: could not compile `as-if-std` (lib) due to previous error
Error: Process completed with exit code 101.
```

Apparently the `#[allow(unused_imports)]` [on the re-export](https://github.com/rust-lang/backtrace-rs/blob/e9da96eb452aa65e79e2342be700544afe509440/crates/as-if-std/src/lib.rs#L17) is no longer enough to satisfy it.

--- a/crates/as-if-std/src/lib.rs
+++ b/crates/as-if-std/src/lib.rs
@@ -9,6 +9,7 @@ extern crate alloc;
 
 // We want to `pub use std::*` in the root but we don't want `std` available in
 // the root namespace, so do this in a funky inner module.
+#[allow(unused_imports)]
 mod __internal {
     extern crate std;
     pub use std::*;

